File: test/integration/ruby/testdata/simple.rb
$file_handle = nil

def say_hi1
  puts 'hi'
  $file_handle = File.open "/"
end

def e
  say_hi1
end

def d
  e
end

def c
  d
end

def b
  c
end

def a
  b
end

def say_hi2
  puts 'hi2'
  $file_handle&.close
end

def c2
  say_hi2
end

def b2
  c2
end

def a2
  b2
end

$stdout.sync = true
puts "PID: #{Process.pid}"

while true
  a
  sleep 0.05
  a2
  sleep 0.05
end

File: test/integration/python/testdata/simple.py
import time
file_handle = None


def say_hi1():
    print('hi')
    global file_handle
    file_handle = open("/")


def e():
    say_hi1()


def d():
    e()


def c():
    d()


def b():
    c()


def a():
    b()


def say_hi2():
    print('hi2')
    if file_handle:
        file_handle.close()


def c2():
    say_hi2()


def b2():
    c2()


def a2():
    b2()


print(f"PID: {os.getpid()}")

while True:
    a()
    time.sleep(0.05)
    a2()
    time.sleep(0.05)
